QuantConnect.TemplateBrokerage.Tests/TemplateFeatureRegressionAlgorithm.cs

```diff
@@ -0,0 +1,559 @@
+                            (security as FutureOption).SetFilter(filter);
+                            break;
+                        case SecurityType.IndexOption:
+                            (security as IndexOption).SetFilter(filter);
+                            break;
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Once the future, option, index option and future option contracts have been set,
+        /// this method will call `BrokerageAlgorithmSettings.InitializeSymbols()` method.
+        /// That method will define the securities allowed per order type according to the 
+        /// brokerage
+        /// </summary>
+        protected virtual void SetupSymbols()
+        {
+            BrokerageAlgorithmSettings.InitializeSymbols();
+            _orderTypes = _orderTypeMethods.Keys.ToList();
+            _pointsFoundPerSymbol = BrokerageAlgorithmSettings.Symbols.ToDictionary(x => x, x => 0);
+            _symbolsHaveBeenSetup = true;
+        }
+
+        public override void OnData(Slice slice)
+        {
+            /// If this brokerage allows futures, wait until getting one future contract
+            if (!SetFutureContract(slice))
+            {
+                Debug($"{Time}: Waiting for future contract to be set...");
+                return;
+            }
+
+            /// If this brokerage allows options, wait until getting one option contract
+            if (!SetOptionContract(BrokerageAlgorithmSettings.CanonicalOptionSymbol, slice.OptionChains.Values, SecurityType.Option, ref BrokerageAlgorithmSettings.OptionContract))
+            {
+                Debug($"{Time}: Waiting for option contract to be set...");
+                return;
+            }
+
+            /// If this brokerage allows index options, wait until getting one index option contract
+            if (!SetOptionContract(BrokerageAlgorithmSettings.CanonicalIndexOptionSymbol, slice.OptionChains.Values, SecurityType.IndexOption, ref BrokerageAlgorithmSettings.IndexOptionContract))
+            {
+                Debug($"{Time}: Waiting for index option contract to be set...");
+                return;
+            }
+
+            /// If this brokerage allows future options, wait until getting one future option contract
+            if (!SetOptionContract(BrokerageAlgorithmSettings.CanonicalFutureOptionSymbol, slice.OptionChains.Values, SecurityType.FutureOption, ref BrokerageAlgorithmSettings.FutureOptionContract))
+            {
+                Debug($"{Time}: Waiting for future option contract to be set...");
+                return;
+            }
+
+            /// Define the securities to test per order type
+            if (!_symbolsHaveBeenSetup)
+            {
+                SetupSymbols();
+            }
+
+            /// Check each symbol allowed by the brokerage has already a price
+            foreach (var symbol in BrokerageAlgorithmSettings.Symbols)
```

Comment:
nit- same as above, maybe extract all these checks to private methods. This one would look like `bool AllSecuritiesHavePrice()`